backend/planty/dashboard/views.py

```diff
@@ -290,3 +295,226 @@ def post(self, request: Request):
+
+        try:
+            instruction: Instruction = Instruction.objects.get(id=kwargs['uuid'])
+        except Model.DoesNotExist:
+            return Response(data={
+                'id': ['Plant with the given ID does not exist']
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        if instruction.user != user:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        if instruction.public:
+            return Response(data={
+                'id:': ['Instruction with this ID is public. Cannot modify public instructions.']
+            }, status=status.HTTP_405_METHOD_NOT_ALLOWED)
+
+
+        instruction.name = data.get('name', instruction.name),
+        instruction.species = data.get('species', instruction.species),
+
+        instruction.fertilizing = data.get('fertilizing', instruction.fertilizing),
+        instruction.insolation = data.get('insolation', instruction.insolation),
+        instruction.watering = data.get('watering', instruction.watering),
+
+        instruction.save()
+
+        return Response(f'Success, instruction {pk} modified.', status=status.HTTP_200_OK)
+
+    def delete(self, request: Request):
+        user: User = request.user
+        serializer = InstructionDeleteSerializer(data=self.kwargs['uuid'])
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            instruction: Instruction = Instruction.objects.get(id=self.kwargs['uuid'])
+        except Model.DoesNotExist:
+            return Response(data={
+                'id': ['Instruction with the given ID does not exist']
+            }, status=status.HTTP_204_NOT_FOUND)
+
+        if instruction.user != user:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        plant.delete()
+        return Response(status=status.HTTP_200_OK)
+
+
+class PopularInstructionsView(APIView):
+    def get(self, request: Request):
+        if request.method == 'GET' and 'limit' in request.GET:
+            N = request.GET['limit']
+        else:
+            N = 10
+
+        suggested = Instruction.objects.filter(public=True)
+        desc_suggested_instruction = suggested.order_by('num_selected').reverse()[:N]
+
+        desc_suggested_instructions_json = []
+        for instruction in desc_suggested_instruction:
+            instruction_json = {
+                'id': str(instruction.id),
+                'name': instruction.name,
+                'species': instruction.species,
+
+                'watering': instruction.watering,
+                'insolation': instruction.insolation,
+                'fertilizing': instruction.fertilizing,
+
+                'num_selected': instruction.num_selected
+            }
+            desc_suggested_instructions_json.append(instruction_json)
+
+        return Response(desc_suggested_instructions_json, status=status.HTTP_200_OK)
+
+
+class SelectInstructionView(APIView):
```

Comment:
Nie jestem pewien co robi to. Rozumiem, że to niby obsługuje wybór intrukcji. Ale czy to nie miałoby być bardziej związane z konkretną rośliną, niż z samą instrukcją?

Reply:
okej, to w takim razie źle zrozumiałem tego select'a
usunąłem endpointa i klasy